src/adaptor/ini.rs

```diff
@@ -0,0 +1,502 @@
+
+use nom::{IResult, space, alphanumeric, multispace};
+
+use super::{Adaptor, Config, AdaptorError};
+
+use std::collections::{HashMap, hash_map};
+use std::fmt::Debug;
+use std::io::{Read, Write};
+use std::iter::FromIterator;
+use std::str;
+
+/// The adaptor struct for INI files
+/// Later, this might contain parameters for the myriad INI quirks
+pub struct IniAdaptor {
+    
+}
+
+impl IniAdaptor {
+    /// Constructs a new `IniAdaptor`
+    pub fn new() -> IniAdaptor {
+        IniAdaptor { }
+    }
+}
+
+impl<'a> Adaptor<'a> for IniAdaptor {
+    /// Deserialize the INI data into the `Config` AST
+    fn deserialize<R>(&self, mut reader: R) -> Result<Config, AdaptorError> 
+        where R: Read 
+    {
+        let mut buffer = Vec::new();
+        reader.read_to_end(&mut buffer);
+
+        // parse the basic INI structure
+        let output = match sections(&buffer) {
+            IResult::Done(_, o) => o,
+            _ => return Err("unable to parse INI data"),
+        };
+
+        let mut combined = HashMap::new();
```

Comment:
It uses a BTreeMap by default and can use a LinkedHashMap if you set the right cargo feature. I'm not sure if we care about it or not :) I tend to err towards ordered maps though, just for determinism..